src/Nethermind/Nethermind.Mev.Test/MevRpcModuleTests.cs

```diff
@@ -309,32 +564,100 @@ public async Task Should_include_bundles_by_eligible_gas_fee()
             var chain = await CreateChain(2);
             chain.GasLimitCalculator.GasLimit = 10_000_000;
 
-            Transaction poolAndBundleTx = Build.A.Transaction.WithGasLimit(GasCostOf.Transaction).WithGasPrice(150ul).SignedAndResolved(TestItem.PrivateKeyC).TestObject;
-            Transaction expensiveBundleTx = Build.A.Transaction.WithGasLimit(GasCostOf.Transaction).WithGasPrice(130ul).SignedAndResolved(TestItem.PrivateKeyA).TestObject;
-            Transaction middleBundleTx = Build.A.Transaction.WithGasLimit(GasCostOf.Transaction).WithGasPrice(120ul).SignedAndResolved(TestItem.PrivateKeyB).TestObject;
-            Transaction cheapBundleTx = Build.A.Transaction.WithGasLimit(GasCostOf.Transaction).WithGasPrice(95ul).SignedAndResolved(TestItem.PrivateKeyB).TestObject;
+            BundleTransaction poolAndBundleTx = Build.A.TypedTransaction<BundleTransaction>()
+                .WithGasLimit(GasCostOf.Transaction)
+                .WithGasPrice(150ul)
+                .SignedAndResolved(TestItem.PrivateKeyC).TestObject;
+            
+            BundleTransaction expensiveBundleTx = Build.A.TypedTransaction<BundleTransaction>()
+                .WithGasLimit(GasCostOf.Transaction)
+                .WithGasPrice(130ul)
+                .SignedAndResolved(TestItem.PrivateKeyA).TestObject;
+            
+            BundleTransaction middleBundleTx = Build.A.TypedTransaction<BundleTransaction>()
+                .WithGasLimit(GasCostOf.Transaction)
+                .WithGasPrice(120ul)
+                .SignedAndResolved(TestItem.PrivateKeyB).TestObject;
+            
+            BundleTransaction cheapBundleTx = Build.A.TypedTransaction<BundleTransaction>()
+                .WithGasLimit(GasCostOf.Transaction)
+                .WithGasPrice(95ul)
+                .SignedAndResolved(TestItem.PrivateKeyB).TestObject;
   
             await SendSignedTransaction(chain, poolAndBundleTx);
 
-            chain.SendBundle(1, poolAndBundleTx, cheapBundleTx);
-            chain.SendBundle(1, expensiveBundleTx);
-            chain.SendBundle(1, middleBundleTx);
+            SuccessfullySendBundle(chain, 1, poolAndBundleTx, cheapBundleTx);
+            SuccessfullySendBundle(chain, 1, expensiveBundleTx);
+            SuccessfullySendBundle(chain, 1, middleBundleTx);
             
             await chain.AddBlock(true);
             
             GetHashes(chain.BlockTree.Head!.Transactions).Should().Equal(GetHashes(new[] { expensiveBundleTx, middleBundleTx, poolAndBundleTx }));
         }
+        
+        [Test]
```

Comment:
In 1559 we have additional consensus checks:
			# The total must be the larger of the two
			assert transaction.max_fee_per_gas >= transaction.max_priority_fee_per_gas
			# the signer must be able to afford the transaction
			assert signer.balance >= transaction.gas_limit * transaction.max_fee_per_gas
https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md
Can we add tests for this cases?

Reply:
@kristofgazso